mmcls/structures/multi_task_data_sample.py

```diff
@@ -26,64 +21,83 @@ def format_task_label(value: Dict, tasks: List[str]) -> LabelData:
 
-    def __init__(self, tasks=None):
-        super(MultiTaskDataSample, self).__init__()
-        self.tasks = tasks
-
-    def set_gt_label(self, value: Dict) -> 'MultiTaskDataSample':
-        """Set label of ``gt_label``."""
-        label = format_task_label(value, self.tasks)
-        if 'gt_label' in self:
-            self.gt_label.label = label.label
+    def set_gt_task(self, value: Dict) -> 'MultiTaskDataSample':
+        """Set label of ``gt_task``."""
+        label = format_task_label(value, self.metainfo)
+        if 'gt_task' in self:
+            self.gt_task.label = label.label
         else:
-            self.gt_label = label
+            self.gt_task = label
         return self
 
-    def set_pred_label(self, value: Dict) -> 'MultiTaskDataSample':
-        """Set label of ``pred_label``."""
-        if 'pred_label' in self:
-            self.pred_label.score = value
+    """
+    def set_pred_task(self, value: Dict) -> 'MultiTaskDataSample':
+        if 'pred_task' in self:
+            self.pred_task.score = value
         else:
-            self.pred_label = LabelData(score=value)
+            self.pred_task = LabelData(score=value)
         return self
-
+    """
     def get_task_mask(self, task_name):
-        return task_name in self.gt_label
+        return task_name in self.gt_task.label
 
+    """
     def get_task_sample(self, task_name):
-        label = getattr(self.gt_label, task_name)
-        label_task = ClsDataSample().set_gt_label(label)
+        label = self.gt_task.label[task_name]
+        label_task = ClsDataSample().set_gt_task(label)
         return label_task
+    """
 
     @property
-    def gt_label(self):
-        return self._gt_label
+    def gt_task(self):
+        return self._gt_task
 
-    @gt_label.setter
-    def gt_label(self, value: LabelData):
-        self.set_field(value, '_gt_label', dtype=LabelData)
+    @gt_task.setter
+    def gt_task(self, value: LabelData):
+        self.set_field(value, '_gt_task', dtype=LabelData)
 
-    @gt_label.deleter
-    def gt_label(self):
-        del self._gt_label
+    @gt_task.deleter
+    def gt_task(self):
+        del self._gt_task
 
+    """
     @property
-    def pred_label(self):
-        return self._pred_label
+    def pred_task(self):
+        return self._pred_task
+
+    @pred_task.setter
+    def pred_task(self, value: LabelData):
+        self.set_field(value, '_pred_task', dtype=LabelData)
+
+    @pred_task.deleter
+    def pred_task(self):
+        del self._pred_task
+    """
+    def to_cls_data_samples(self, task_name):
+        label = self.gt_task.label[task_name]
+        label_task = ClsDataSample(
+            metainfo=self.metainfo.get(task_name, {})).set_gt_label(
+                value=label)
+        return label_task
+
+    def to_multi_task_data_sample(self, task_name):
+        label = self.gt_task.label[task_name]
+        label_task = MultiTaskDataSample().set_gt_task(value=label)
```

Comment:
please set also values from `self.pred_task[task_name]` into the output data_sample